pancancer_evaluation/mutation_prediction.py

```diff
@@ -0,0 +1,465 @@
+                       num_folds=num_folds, fold_no=fold_no,
+                       use_pancancer=use_pancancer, seed=self.seed)
+            except ValueError:
+                raise NoTestSamplesError(
+                    'No test samples found for cancer type: {}, '
+                    'gene: {}\n'.format(cancer_type, gene)
+                )
+
+            y_train_df = self.y_df.reindex(X_train_raw_df.index)
+            y_test_df = self.y_df.reindex(X_test_raw_df.index)
+
+            X_train_df, X_test_df = self._preprocess_data(X_train_raw_df, X_test_raw_df)
+
+            try:
+                # also ignore warnings here, same deal as above
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    model_results = train_model(
+                        x_train=X_train_df,
+                        x_test=X_test_df,
+                        y_train=y_train_df,
+                        alphas=cfg.alphas,
+                        l1_ratios=cfg.l1_ratios,
+                        seed=self.seed,
+                        n_folds=cfg.folds,
+                        max_iter=cfg.max_iter
+                    )
+                    (cv_pipeline,
+                     y_pred_train_df,
+                     y_pred_test_df,
+                     y_cv_df) = model_results
+            except ValueError:
+                raise OneClassError(
+                    'Only one class present in test set for cancer type: {}, '
+                    'gene: {}\n'.format(cancer_type, gene)
+                )
+
+            # get coefficients
+            coef_df = extract_coefficients(
+                cv_pipeline=cv_pipeline,
+                feature_names=X_train_df.columns,
+                signal=signal,
+                seed=self.seed
+            )
+            coef_df = coef_df.assign(gene=gene)
+            coef_df = coef_df.assign(fold=fold_no)
+
+            try:
+                # also ignore warnings here, same deal as above
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    metric_df, gene_auc_df, gene_aupr_df = self._get_metrics(
+                        y_train_df, y_test_df, y_cv_df, y_pred_train_df,
+                        y_pred_test_df, gene, cancer_type, signal, fold_no
+                    )
+            except ValueError:
+                raise OneClassError(
+                    'Only one class present in test set for cancer type: {}, '
+                    'gene: {}\n'.format(cancer_type, gene)
+                )
+
+            self.results['gene_metrics'].append(metric_df)
+            self.results['gene_auc'].append(gene_auc_df)
+            self.results['gene_aupr'].append(gene_aupr_df)
+            self.results['gene_coef'].append(coef_df)
+
+        self._save_results(gene, cancer_type, signal)
+        del self.results
+
+
+    def _get_metrics(self,
```

Comment:
This is a great idea for a function and I need to add something like this to my code